guava/src/com/google/common/collect/TreeMultiset.java

```diff
@@ -801,14 +794,7 @@ public int count(Comparator<? super E> comparator, E e) {
 
         right = initRight.setCount(comparator, e, expectedCount, newCount, result);
 
-        if (result[0] == expectedCount) {
-          if (newCount == 0 && result[0] != 0) {
-            this.distinctElements--;
-          } else if (newCount > 0 && result[0] == 0) {
-            this.distinctElements++;
-          }
-          this.totalCount += newCount - result[0];
-        }
+        calculatedQuantity(expectedCount, newCount, result);
         return rebalance();
```

Comment:
It looks like instead of pulling out a method, you might as well just pull this out of the if?

Reply:
Thank you for your advice!

Reply:
I modified my code with reference to your suggestion.

Reply:
I modified my code based on your suggestions.

Reply:
@lowasser I modified my code based on your suggestions.Thanks